### PR TITLE
[Main] Add Cloud Message to Personal server/desktop

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Implements a message bar at the top of the Personal Server/Desktop editions. It will display 24 hours after the user is created. And will continue to display until the user closes it. At this point it will not display again.

#### Ticket Link
https://community.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/b4ctmm917q3rzdfi451qtuxjzcw/vch9s93uimjrfik7fisjczzpfkc/ciwthmxxn1by8jcrgr8trk1g8ge

#### Screenshot

Screen Shot 2022-05-09 at 12 50 26 PM

### CODE REVIEW

File: webapp/src/components/messages/cloudMessage.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.CloudMessage {
```
Comment (Pinjasaur) [inline webapp/src/components/messages/cloudMessage.scss:1]: I assume this file needs the copyright comment banner at the top?

File: webapp/src/components/messages/cloudMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import {useAppSelector, useAppDispatch} from '../../store/hooks'
+import octoClient from '../../octoClient'
+import {IUser, UserConfigPatch} from '../../user'
+import {getMe, patchProps, getCloudMessageCanceled} from '../../store/users'
+
+import CompassIcon from '../../widgets/icons/compassIcon'
+import TelemetryClient, {TelemetryCategory, TelemetryActions} from '../../telemetry/telemetryClient'
+
+import './cloudMessage.scss'
+const signupURL = 'http://mattermost.com/pricing'
```
Comment (Pinjasaur) [inline webapp/src/components/messages/cloudMessage.tsx:22]: nit: I think we want this to be `https://`

File: webapp/src/components/messages/cloudMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            const patchedProps = await octoClient.patchUserConfig(me.id, patch)
+            if (patchedProps) {
+                dispatch(patchProps(patchedProps))
+            }
+        }
+    }
+
+    if (me) {
+        const installTime = Date.now() - me.create_at
+        if (Utils.isFocalboardPlugin() || cloudMessageCanceled || installTime < displayAfter) {
```
Comment (Pinjasaur) [inline webapp/src/components/messages/cloudMessage.tsx:53]: question: for the logic in the nested `if`, wouldn't you want to check `Utils.isFocalboardPlugin() || cloudMessageCanceled` irrespective of `if (me)`?

perhaps something like:


```ts
if (Utils.isFocalboardPlugin() || cloudMessageCanceled) {
    return null
}

if (me) {
    const installTime = Date.now() - me.create_at
    if (installTime < displayAfter) {
        return null
    }
}
```

File: webapp/src/components/messages/cloudMessage.scss (hunk 1 above)
Comment (sbishel) [reply webapp/src/components/messages/cloudMessage.scss:1]: I added it, but I really don't know. Not very many scss files have it.

File: webapp/src/components/messages/cloudMessage.tsx (hunk 3 above)
Comment (sbishel) [reply webapp/src/components/messages/cloudMessage.tsx]: I debated...it really doesn't matter as me should never be undefined, but I do think it makes it more readable and understandable, so I updated.

**Review by asaadmahmood — APPROVED** [review]
LGTM

**Review by Pinjasaur — APPROVED** [review]
LGTM

**Pull request merged**